### DISCUSSION TITLE
Multiple nested queries

### DISCUSSION THREAD

**Discussion by wanjas** [prisma/prisma#7388] [Q&A] [answered]
Hello, 

Prisma **does** allow applying multiple nested queries to the same entity as in the example below. 
Sometimes it makes sense to first update/delete an existing nested record and create a new one. 
The following example **works** exactly the way I intended. It deletes/updates existing records and then adds one new. 

But I can't find any confirmation on how nested queries execution is ordered (if it is). 
Sorry, if I failed to find already existing information. 

It would be really nice to have this use case to be either strictly described in docs and supported or be explicitly forbidden. 
(Support is of course better because it's handy and kinda works already :smile_cat: )

``` javascript
await prismaClient.user.update({
  where: {
    id: userId,
  },
  data: {
    userRecords: {
      deleteMany: { /* delete all */  },
     /* or */ 
     // updateMany: { where: {}, data: { isActive: false }  } 
      create: {
           content: 'the only or the only active record',
           isActive: true,
      },
   },
 },
});
```

**Comment by ryands17** [comment, marked as answer]
@wanjas 👋 

As per the latest ECMAScript standard, object keys are run in the order of what you specify, so in the above example `deleteMany` will run before `create`.

I agree this should be documented. Thanks for bringing this up 👍

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
